src/ompl/util/src/RandomNumbers.cpp

```diff
@@ -217,3 +238,297 @@ void ompl::RNG::eulerRPY(double value[3])
     value[1] = acos(1.0 - 2.0 * uni_()) - boost::math::constants::pi<double>() / 2.0;
     value[2] = boost::math::constants::pi<double>() * (-2.0 * uni_() + 1.0);
 }
+
+
+void ompl::RNG::uniformNormalVector(unsigned int n, double value[])
+{
+    // Typedef the container type for the distribution and generator
+    typedef boost::numeric::ublas::shallow_array_adaptor<double> container_type_t;
+
+    // Construct the Boost distribution and generator. These would ideally be at the class level, but the spherical distribution requires dimension at construction.
+    // Tell them to return the type double in a shallow_array_adaptor container:
+    boost::uniform_on_sphere<double, container_type_t> uniformSphereDist(n);
+    boost::variate_generator<boost::mt19937&, boost::uniform_on_sphere<double, container_type_t> > normalizedVector(generator_, uniformSphereDist);
```

Comment:
I believe some of these objects are quite expensive to construct, so we should cache them. Presumably there would not be a different n for every call, so maybe store these in a vector<int> to boost::unique ptrs? Then we just do an index operation + if statement on whether the object was already constructed. In the header file, that could be an unimplemented struct (pimpl pattern) and have it implemented in the .cpp only for information hiding.